### PR TITLE
Initial implementation

### PR DESCRIPTION
**Pull request by lennym**
Polls for pending export requests and builds assets into S3 before marking requests as complete.

### CODE REVIEW

File: lib/rops/index.js [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+          'projects.title',
+          'projects.status as projectStatus',
+          'projects.issueDate',
+          'projects.expiryDate',
+          'projects.revocationDate',
+          'licenceHolder.firstName',
+          'licenceHolder.lastName',
+          'licenceHolder.email',
+          'licenceHolder.telephone',
+          'rops.*'
```
Comment (lennym) [inline lib/rops/index.js:58]: This _might_ want to be reduced down to only the fields that relate to the ROP submission and exclude the fields only used to configure procedure options. Out of scope for now though.

File: lib/rops/index.js [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+        )
+        .joinRelated('licenceHolder')
+        .leftJoinRelated('rops')
+        .whereRopsDue(key)
+        .where(q => q.where('rops.year', key).orWhereNull('rops.year'))
```
Comment (wheelsandcogs) [inline lib/rops/index.js:63]: What does this add on top of the `whereRopsDue(year)` ?

Comment (lennym) [reply lib/rops/index.js:63]: My understanding of `.whereRopsDue(year)` is that it filters the _projects_ table to ones with a ROP due in that year. This is then adding an extra filter to the `rops` associated with those projects.

Without this, you potentially get multiple rows for each project, with one per existing ROP for any year as long as the project itself was due a ROP in that year.

Comment (wheelsandcogs) [reply lib/rops/index.js:63]: I think there's a `withRops(year)` query builder function on Projects schema, that might do what you're after as well.

Comment (wheelsandcogs) [reply lib/rops/index.js:63]: Although that won't include null year, so maybe not.

Comment (lennym) [reply lib/rops/index.js:63]: I'll check it out. As long as it doesn't mind if there is no ROP at all then it looks like the guy.

Comment (lennym) [reply lib/rops/index.js:74]: It needs to be as it is above. Using `.withRops` caused multiple rows for a single project when there were multiple ROPs for that project across different years.

**Pull request merged**